### PR TITLE
distlib: build executables from source

### PR DESCRIPTION
**Pull request by naveen521kk**
Possibly it should be done for pip also
Fixes: https://github.com/msys2/MINGW-packages/issues/7607

### CODE REVIEW

**Comment by naveen521kk** [comment]
@lazka How can we name the launcher for arm?

**Comment by lazka** [comment]
I'd just name it the same for now. Otherwise they wont be used.

File: mingw-w64-python-distlib/PKGBUILD [hunk 1]
```diff
@@ -48,8 +30,39 @@ prepare() {
...
+  local _gui_lflags="-lshlwapi"
+  local _cli_lflags="-lshlwapi"
+  if check_option "strip" "y"; then
+    _gui_cflags+=" -s"
+    _cli_cflags+=" -s"
+  fi
+  case "$MSYSTEM" in
+  MINGW*)
+    _gui_cflags+=" -mcrtdll=msvcr120"
+    _cli_cflags+=" -mcrtdll=msvcr120"
```
Comment (lazka) [inline mingw-w64-python-distlib/PKGBUILD:47]: What's this about?

Comment (naveen521kk) [reply mingw-w64-python-distlib/PKGBUILD:47]: it fails without them. https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/dupenv-s-wdupenv-s?view=msvc-160
google got me to https://stackoverflow.com/a/61556667/12858827

**Comment by naveen521kk** [comment]
> Otherwise they won't be used.

It's simple enough to patch. But I can name it the same as `t64` also, if we could decide on a name.

File: mingw-w64-python-distlib/PKGBUILD (hunk 1 above)
Comment (lazka) [reply mingw-w64-python-distlib/PKGBUILD:47]: ok

**Comment by naveen521kk** [comment]
ready!

**Comment by jeremyd2019** [comment]
What I had done with setuptools is https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-python-setuptools/0007-windows-arm64.patch but I only did that during the time when it tried to include *all* arches in the package, before that I just overloaded the 64 variant as you are talking about here.

**Comment by naveen521kk** [comment]
If you give a nice executable name I can patch like that, or this would anyway work.

File: mingw-w64-python-distlib/PKGBUILD [hunk 2]
```diff
@@ -48,8 +30,43 @@ prepare() {
...
+    gcc $_cli_cflags -o distlib/t32.exe PC/launcher.c PC/resources.res $_cli_lflags
+    gcc $_gui_cflags -o distlib/w32.exe PC/launcher.c PC/resources.res $_gui_lflags
+    ;;
+  x86_64-w64-mingw32)
+    gcc $_cli_cflags -o distlib/t64.exe PC/launcher.c PC/resources.res $_cli_lflags
+    gcc $_gui_cflags -o distlib/w64.exe PC/launcher.c PC/resources.res $_gui_lflags
+    ;;
+  aarch64-w64-mingw32)
+    gcc $_cli_cflags -o distlib/t64.exe PC/launcher.c PC/resources.res $_cli_lflags
+    gcc $_gui_cflags -o distlib/w64.exe PC/launcher.c PC/resources.res $_gui_lflags
```
Comment (mati865) [inline mingw-w64-python-distlib/PKGBUILD:63]: All these should probably use `cc`. Clang subsystems use `gcc` as `clang` but hopefully it's temporary measure.

Comment (naveen521kk) [reply mingw-w64-python-distlib/PKGBUILD]: ok done.

**Comment by naveen521kk** [comment]
@lazka can this be merged?

**Pull request merged**